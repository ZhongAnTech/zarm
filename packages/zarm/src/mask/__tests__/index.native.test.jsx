import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TouchableWithoutFeedback } from 'react-native';
import Mask from '../index.native';

describe('Mask', () => {
  it('renders correctly', () => {
    const wrapper = render(<Mask visible />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('type is transparent', () => {
    const wrapper = render(<Mask visible type="transparent" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Mask visible onClick={onClick} />);
    wrapper.find(TouchableWithoutFeedback).props().onPress();
    expect(onClick).toBeCalled();
  });
});
