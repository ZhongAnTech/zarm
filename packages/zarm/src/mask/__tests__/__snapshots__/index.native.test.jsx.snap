// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mask renders correctly 1`] = `
<react-native-mock>
  <react-native-mock />
</react-native-mock>
`;

exports[`Mask type is transparent 1`] = `
<react-native-mock>
  <react-native-mock />
</react-native-mock>
`;
